mlflow/tracking/_tracking_service/client.py

```diff
@@ -631,8 +646,7 @@ def _record_logged_model(self, run_id, mlflow_model):
             )
         self.store.record_logged_model(run_id, mlflow_model)
 
-    def _get_artifact_repo_for_trace(self, request_id):
-        trace_info = self.get_trace_info(request_id)
```

Comment:
This significantly hurts performance because it means we end up calling `GetTraceInfo` unnecessarily for every trace during search. This is unnecessary because the search REST API call already returns the trace infos.